test_framework/terraform/aws/talos/main.tf

```diff
@@ -277,3 +277,89 @@ data "talos_cluster_health" "this" {
   control_plane_nodes  = aws_instance.lh_aws_instance_controlplane.*.private_ip
   worker_nodes         = aws_instance.lh_aws_instance_worker.*.private_ip
 }
+
+# Generate Talos Schematic and upload to Talos Factory
+resource "null_resource" "upload_schematic" {
+  depends_on = [
+    data.talos_cluster_health.this,
+    local_file.talosconfig
+  ]
+
+  provisioner "local-exec" {
+    command = <<EOT
+      if [ ! -f "${path.module}/longhorn-talos.yaml" ]; then
+        echo "Error: longhorn-talos.yaml file not found"
+        exit 1
+      fi
+
+      curl -X POST --data-binary @${path.module}/longhorn-talos.yaml https://factory.talos.dev/schematics > ${path.module}/schematic_response.json
+    EOT
+  }
+}
+
+# Extract schematic ID from uploaded response
+data "local_file" "schematic_response" {
+  depends_on = [null_resource.upload_schematic]
+  filename   = "${path.module}/schematic_response.json"
+}
+
+locals {
+  schematic_id = jsondecode(data.local_file.schematic_response.content).id
+  talos_version = "v${var.os_distro_version}"
+}
+
+# Upgrade Talos cluster using schematic ID
+resource "null_resource" "upgrade_talos_cluster" {
+  depends_on = [
+    data.local_file.schematic_response,
+    local_file.talosconfig,
+    data.talos_cluster_health.this
+  ]
+
+  provisioner "local-exec" {
+    command = <<EOT
+      if [ ! -f "${path.module}/talos_k8s_config" ]; then
+        echo "Error: talos_k8s_config file not found"
+        exit 1
+      fi
+
+      ENDPOINT="${aws_instance.lh_aws_instance_controlplane[0].public_ip}"
+      CONTROL_PLANE_NODES=$(echo "${join(",", aws_instance.lh_aws_instance_controlplane.*.private_ip)}")
+
+      talosctl --talosconfig ${abspath(path.module)}/talos_k8s_config \
```

Comment:
What's the difference between this `talosctl` and the one in `upgrade_controlplane_nodes`?